### PR TITLE
Rebrand to JsonPath4K and update dependencies

### PR DESCRIPTION
**Pull request by JesusMcCloud**
Nomen est Omen.

Release 2.2.0:
- Rebranding to JsonPath4K
  - change Maven coordinates to `at.asitplus:jsonpath4k`
  - publish relocation POM for Version 2.0.0
- Dependency Updates
  - Update to Kotlin 2.0.0
  - Update to Kotest 5.9.1
  - Update to kotlinx-serialization 1.7.1
  - Gradle 8.9
  - Antlr-Kotlin 1.0.0-RC4

### CODE REVIEW

**Comment by JesusMcCloud** [comment]
repo will be renamed right after this PR it through

File: jsonpath4k/build.gradle.kts [hunk 1]
```diff
@@ -30,6 +27,29 @@ repositories {
...
+val generateKotlinGrammarSource = tasks.register<AntlrKotlinTask>("generateKotlinGrammarSource") {
+    dependsOn("cleanGenerateKotlinGrammarSource")
+
+    // compiling any *.g4 files within the project
+    source = fileTree(layout.projectDirectory) {
+        include("**/*.g4")
+    }
+
+    // We want the generated source files to have this package name
+    val pkgName = "at.asitplus.jsonpath.generated"
```
Comment (nodh) [inline jsonpath4k/build.gradle.kts:39]: seems like a redundant variable

**Review by nodh — APPROVED** [review]

File: jsonpath4k/src/commonMain/kotlin/at/asitplus/jsonpath/implementation/AntlrJsonPathSemanticAnalyzerVisitor.kt [hunk 2]
```diff
@@ -39,29 +39,29 @@ internal class AntlrJsonPathSemanticAnalyzerVisitor(
     }
 
     override fun aggregateResult(
-        aggregate: AbstractSyntaxTree<out JsonPathExpression>?,
+        aggregate: AbstractSyntaxTree<out JsonPathExpression>,
         nextResult: AbstractSyntaxTree<out JsonPathExpression>
     ): AbstractSyntaxTree<out JsonPathExpression> {
```
Comment (acrusage-iaik) [inline jsonpath4k/src/commonMain/kotlin/at/asitplus/jsonpath/implementation/AntlrJsonPathSemanticAnalyzerVisitor.kt:44]: Unsure about this change, but it seems to not yield any issues

Comment (JesusMcCloud) [reply jsonpath4k/src/commonMain/kotlin/at/asitplus/jsonpath/implementation/AntlrJsonPathSemanticAnalyzerVisitor.kt:44]: that was pulled-in by an update to latest antlr

**Pull request merged**